### PR TITLE
(WIP) - UristKitchen - Disease & Spoilage Framework

### PR DESCRIPTION
**Pull request by Laniakeaaaa**
### **UristKitchen - Disease & Spoilage Framework:**

This is WIP, with a bunch of work left to do, this probably won't actually get activated onto food until the rest of UristKitchen is finished. So don't worry and eat all the raw brain you want until then!

**Adds framework for diseases for raw and rotten food, with species restriction and effects.**
**Adds spoilage system for perishable food.**

### **Races:**
**Humans** : Vulnerable to all food-borne illnesses.
**Teshari** : Vulnerable to only rotten food illnesses.
**Unathi** : Vulnerable to only rotten food illnesses.
**Skrell** : Vulnerable to all foodborne illnesses, though they have their own food.
**Vox** : Unaffected by any food illnesses, due to their scavenger nature.
**Dionae** : Only affected by spore/fungal related illnesses from rotten food.

### **Diseases:**
Food is a vector for disease if eaten raw, or if it is rotten, this is dependant on your species as carnivorous species, such as Unathi and Teshari are able to eat raw meat without issues. Food will have a small chance, when eaten raw, or higher when rotten to give you a disease, that will be treatable with certain medicine, or through waiting.

These diseases have been designed to mostly be specific to the person who consumed it themself. Meaning that only they will suffer, very few of these diseases are actually contagious. 

Parasitic Diseases will require antiparasitic drugs that chemistry will be able to produce.

### **Disease Types:**

**Diseases are split into three categories:** 
**Bacteria,** such as E. Coli, Salmonella, Listeria, Etc. (These will only appear in raw food/rotten food.)
**Parasites,** such as Tapeworms and Alien Gut Worms. (These will only appear in meats.)
**Fungal Diseases,** such as spores/molds. (These will only appear in rotten food.)

### Current Diseases:
**Bacteria:**
E. Coli Food Poisoning: Food poisoning, vomiting and so on.
Listeria: More dangerous food poisoning, causing severe stomach pains.
Salmonella: Food Poisoning causes fluid loss, stomach pains, and nausea.
Rot Sprawl: Nausea, vomiting and leading to slow intake of toxin damage.

**Spores/Fungi:**
Mycrosohyosi: Rotten Food poisoning, causes lung pain, shortness of breath and coughing up spores.

**Parasites:**
Parasitic Infection: Causes hunger pangs, and leaves you unable to fill your nutrition bar.
Gutworms: Slowly damage your internal organs. (Debating not really having these, as they are OP)

### To-Do:

- [ ] Attach Dependancies to Food System, such as with meat.
- [x] Port over Navi's Cooking Sprites for Fridge.
- [ ] Allow for refrigeration and spoiling of meat and other perishable products.
- [ ] Add various medications to negate the effects.
- [ ] Strongly consider balancing and griefing issues caused by this.
- [ ] Add Antiparasitics and the support to cure the disease using these.
- [x] Consider other racial supports for food and illnesses.
- [ ] Add preservatives reagents for food.
- [ ] Allow for spoilage to update examine desc, change icon, etc.
- [ ] Optimize a way for rotting to work in ticks, or reset when refrigerated?
- [ ] Set a FLAG state for Perishable and Non-Perishable foods.

### CODE REVIEW

File: code/modules/urist/datums/uristvirus2.dm [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+	name = "Nauseous Syndrome"
+	stage = 2
+	badness = VIRUS_COMMON
+	delay = 30
+	activate(var/mob/living/carbon/human/mob,var/multiplier)
+		to_chat(mob, "<span class='warning'>Your head is spinning and you feel nauseous.</span>")
+
+// Sickly
+/datum/disease2/effect/vomiting	// Causes the person to vomit, gross.
+	name = "Sickly Syndrome"
```
Comment (scrdest) [inline code/modules/urist/datums/uristvirus2.dm]: A bit non-indicative IMO. 'Emesis Syndrome'?

File: code/modules/urist/datums/uristvirus2.dm [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+
+// Virus Effects defined below.
+// Make sure that they are neatly organized, I've seperated them into seperate commented areas, for my own sanity - Shippy
+// TODO See about adding some of these effects into normal virus generation, if they aren't already...
+
+													// Stomach Related Food Poisoning \\
+
+// Stomach Gurgling
+/datum/disease/effect/stomachbad	// Stomach unsettled.
+	name = "Stomach Unsettled Syndrome"
```
Comment (scrdest) [inline code/modules/urist/datums/uristvirus2.dm]: 'Gastritis'?

File: code/modules/urist/datums/uristvirus2.dm [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+
+/datum/disease2/disease/salmonella/New()
+	..()
+	var/datum/disease2/effect/stomach/E1 = new()	// Minor Upset Stomach, Stage 1.
+	E1.stage = 1
+	effects += E1
+	var/datum/disease/effect/dehydration/E2 = new()	// Dehydration, Stage 2.
+	E2.stage = 2
+	effects += E2
+	var/datum/disease/effect/choking/E3 = new()	// Nausea, Stage 3.
```
Comment (scrdest) [inline code/modules/urist/datums/uristvirus2.dm]: Shouldn't this be /effect/nausea?

File: code/modules/urist/datums/uristvirus2.dm [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+	E4.stage = 4
+	effects += E4
+
+// Salmonella: Caused by eating raw meat, certain rotten food, etc. Causes upset stomach, vomitting, annoying messages.
+/datum/disease2/disease/salmonella
+	infectionchance = 0
+	speed = 3
+	spreadtype = "Contact"
+	var/max_stage = 4
+	var/list/affected_species = list(SPECIES_HUMAN,SPECIES_TESHARI,SPECIES_SKREL) // It'd be funny if Vox got Salmonella.
```
Comment (scrdest) [inline code/modules/urist/datums/uristvirus2.dm]: Typo: SPECIES_SKREL -> _SKRELL

File: code/modules/urist/datums/uristvirus2.dm [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+	var/datum/disease2/effect/parasitesgrowth/E1 = new()	// Parasite Growing.
+	E1.stage = 1
+	effects += E1
+	var/datum/disease/effect/parasiteorgan/E2 = new()	// Parasite Small Pain
+	E2.stage = 2
+	effects += E2
+	var/datum/disease/effect/parasitesevere/E3 = new()	// Parasite Severe Pain
+	E3.stage = 3
+	effects += E3
+	var/datum/disease2/effect/organs/E4 = new()	// Infectious Spores Stage 4.	TODO This has serious balance considerations, as this is LETHAL.
```
Comment (scrdest) [inline code/modules/urist/datums/uristvirus2.dm]: The part before TODO seems like an overlooked copy/paste artifact.

File: code/modules/urist/datums/uristvirus2.dm [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+	badness = VIRUS_MILD
+	delay = 90
+	chance_max = 25 // Or you'll be hungry forever.
+	activate(var/mob/living/carbon/human/mob,var/multiplier)
+		to_chat(mob, "<span class='warning'>You feel like you are going to puke...</span>")
+		mob.emote("vomit")
+
+// Stomach paralysis
+/datum/disease/effect/stomachparalysis		// Causes person to fall to floor in pain.
+	name = "Severe Stomach Paralysis Syndrome"
```
Comment (scrdest) [inline code/modules/urist/datums/uristvirus2.dm]: 'Gastroparesis'?

File: code/modules/urist/datums/uristvirus2.dm (hunk 5 above)
Comment (Laniakeaaaa) [reply code/modules/urist/datums/uristvirus2.dm]: Nah, I've been considering whether to keep the organ one, since it's very iffy balance wise.

**Comment by Laniakeaaaa** [comment]
Debating as to what sort of system would work with how perishable food rots, if it's set to a specific tick and then rots, or if placing it in refrigeration would reset the timer and such. I don't want it to be overly punishing to the Chefs.

Considering just having only raw food, such as meat cuts and certain botanical food rot, as it'd be shitty to spend all your time cooking and then it rots five minutes later. https://github.com/vgstation-coders/vgstation13/pull/16118   Had a system for rotting food done with flags and ticks, as well as a fly system, but the flies seemed a little excessive. 

https://docs.google.com/document/d/1gf1ggmNNdYFEE9gbYiZWmEYV8qSgWgDFjxkllsjy99c/edit?usp=sharing
Has some additional data/scope of how this hopefully will pan out.

Additionally my only other concern is how overlays/rotting would be sprited, making seperate sprites would take far too long, grayscale was done in the VGstation commit, which may be a good idea, or relying on overlays, or stink clouds may be a better alternative.

**Comment by Laniakeaaaa** [comment]
Closing for now, stale and I want to come back to this when I'm not busy with IRL shit. On my to-do list, most kitchen sprites are already added.

**Pull request closed without merging**